geopandas/geodataframe.py

```diff
@@ -1459,7 +1467,15 @@ def _constructor(self):
 
     @property
     def _constructor_sliced(self):
-        return _geoseries_constructor_with_fallback
+        def _geodataframe_constructor_sliced(data=None, index=None, crs=None, **kwargs):
+
+            srs = pd.Series(data, index, **kwargs)
+            is_row_proxy = srs.index is self.columns
+            if is_geometry_type(srs) and not is_row_proxy:
+                srs = GeoSeries(srs, crs=crs)
```

Comment:
One more ;) 
I know this is in the original `_geoseries_constructor_with_fallback` as well, but I am wondering if passing the `crs` is ever needed here. In practice, this `_constructor_sliced` typically gets called from places within pandas, so where no `crs` keyword will get passed. And even if it is present, since we only get here if the actual underlying data is a GeometryArray, that should already have the crs? 

Running the tests without passing this here doesn't give any failures for me. 
Now, given this not new but how it was already done before, it's maybe good to leave it for a separate PR (I can do a follow-up one, it might also depend on the pandas version etc)